07.ls_object/display.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Display
+  FILE_TYPE = { 'file' => '-', 'directory' => 'd', 'link' => 'l' }.freeze
+  PERMISSION = { '0' => '---', '1' => '--x', '2' => '-w-', '3' => '-wx', '4' => 'r--', '5' => 'r-x', '6' => 'rw-', '7' => 'rwx' }.freeze
+  COLUMN = 3
+
+  def initialize(file_info_list)
+    @file_info_list = file_info_list
+  end
+
+  def show_files(long: false)
+    if long
+      display_long_output
+    else
+      display_normal_output
+    end
+  end
+
+  private
+
+  def display_long_output
+    max_lengths = calculate_max_lengths(@file_info_list)
+    puts "total #{@file_info_list.sum(&:blocks)}"
+    @file_info_list.each do |info|
+      puts generate_long_format(info, max_lengths)
+    end
+  end
+
+  def calculate_max_lengths(file_info_list)
+    info_lengths = { nlink: [], user: [], group: [], size: [] }
```

Comment:
```suggestion
    max_lengths = { nlink: [], user: [], group: [], size: [] }
```

メソッド名にあわせてこういう名前にした方がいいかも。

あと、Hash.newを使ってデフォルト値を `[]` にするようなコードを書いてみるのはどうでしょうか？

```suggestion
    max_lengths = Hash.new ...
```

Reply:
書き換えや保守をいかに楽にするかで、いろんなコードがあるんですね。
奥深い。。🙌